lint/rule_panel_promql.go

```diff
@@ -25,9 +22,20 @@ func panelHasQueries(p Panel) bool {
 
 // parsePromQL returns the parsed PromQL statement from a panel,
 // replacing eg [$__rate_interval] with [5m] so queries parse correctly.
+// We also replace various other Grafana global variables.
 func parsePromQL(t Target) (parser.Expr, error) {
-	expr := rangeVectorRegexp.ReplaceAllString(t.Expr, "[5m]")
-	expr = subqueryRegexp.ReplaceAllString(expr, "[5m:]")
+	expr := t.Expr
+	for _, pattern := range []struct {
+		variable     string
+		replacesment string
+	}{
+		{"$__range_s", "18000"},
```

Comment:
We can perhaps add this later, but I still have an open PR for adding a `$__rate_interval` rule to the previous repo. it also substitutes known global variables, with deterministic magic/random numbers, so they can be explicitly evaluated later. I.E. I know I can check if a given range vector used `$__rate_interval` by checking if the value is `8869990787ms`.

That PR also provides values for all template variables.

I can add both of those things here in a follow-up PR. I think this is fit for purpose.

```
// querySubs contains a map of "magic number" values for all known global query variables defined in Grafana
// which can be used for substitution such that linting rules can successfully parse the query, and the original
// global variable can still be determined.
//
// see also https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/prometheus.go#L277-L284
//
// Because you're wondering, the numbers are "real" random numbers with 10 digits generated from
// rolling a pair of dice. If the number on the dice was 10, 0 was used, and if it was over 10 that
// roll took two decimal places. I.E. 11 == 11
var querySubs = map[string]string{
	"$__rate_interval": "8869990787ms",
	"$__interval":      "4867856611ms",
	"$__interval_ms":   "7781188786ms",
	"$__range":         "6069770749ms",
	"$__range_ms":      "6737667980ms",
	"$__range_s":       "9397795485ms",
}
```

Reply:
I've added $__interval and made some of them just numbers (the _ms ones aren't time formatted) so we should now be in sync.